test/e2e/common.go

```diff
@@ -284,13 +283,7 @@ func EnsureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyCl
 		InstallCNI(ctx, input, cluster.Spec.ClusterNetwork.Pods.CIDRBlocks, hasWindows)
 	}
 	controlPlane := discoveryAndWaitForControlPlaneInitialized(ctx, input, result)
-	v, err := semver.ParseTolerant(input.ConfigCluster.KubernetesVersion)
```

Comment:
With the new watcher interface, we use new input and result structs that don't have any way to get Kubernetes version. Can we stop worrying about this pre-1.23 rule?

Reply:
Yes, we can stop worrying about the pre-1.23 rules. This was mainly introduced sometime back to test the OOT transition of csi and cloud-provider-azure. 

I am not sure why "not being able to fetch k8s version" is kind enforced now. But we can worry about this when we need it.